sycl/test/basic_tests/vectors/assign.cpp

```diff
@@ -0,0 +1,84 @@
+
+#include <sycl/sycl.hpp>
+
+using sycl::half;
+using sycl::vec;
+using sw_half_1 = decltype(std::declval<vec<half, 4>>().swizzle<1>());
+using sw_half_2 = decltype(std::declval<vec<half, 4>>().swizzle<1, 2>());
+
+using sw_float_1 = decltype(std::declval<vec<float, 4>>().swizzle<1>());
+using sw_float_2 = decltype(std::declval<vec<float, 4>>().swizzle<1, 2>());
+
+using sw_double_1 = decltype(std::declval<vec<double, 4>>().swizzle<1>());
+using sw_double_2 = decltype(std::declval<vec<double, 4>>().swizzle<1, 2>());
+
+#if __INTEL_PREVIEW_BREAKING_CHANGES
+#define NOT_IN_PREVIEW !
+#else
+#define NOT_IN_PREVIEW
+#endif
+
+// clang-format off
+
+static_assert(               std::is_assignable_v<vec<half, 1>, half>);
+static_assert(NOT_IN_PREVIEW std::is_assignable_v<vec<half, 1>, float>);
+static_assert(NOT_IN_PREVIEW std::is_assignable_v<vec<half, 1>, double>);
+static_assert(               std::is_assignable_v<vec<half, 1>, vec<half, 1>>);
+static_assert(NOT_IN_PREVIEW std::is_assignable_v<vec<half, 1>, vec<float, 1>>);
+static_assert(NOT_IN_PREVIEW std::is_assignable_v<vec<half, 1>, vec<double, 1>>);
+static_assert(               std::is_assignable_v<vec<half, 1>, sw_half_1>);
+static_assert(               std::is_assignable_v<vec<half, 1>, sw_float_1>);
+static_assert(               std::is_assignable_v<vec<half, 1>, sw_double_1>);
+static_assert(              !std::is_assignable_v<vec<half, 1>, sw_half_2>);
+static_assert(              !std::is_assignable_v<vec<half, 1>, sw_float_2>);
+static_assert(              !std::is_assignable_v<vec<half, 1>, sw_double_2>);
+
+static_assert(               std::is_assignable_v<vec<half, 2>, half>);
+static_assert(               std::is_assignable_v<vec<half, 2>, float>);
+static_assert(               std::is_assignable_v<vec<half, 2>, double>);
+static_assert(               std::is_assignable_v<vec<half, 2>, vec<half, 1>>);
+static_assert(              !std::is_assignable_v<vec<half, 2>, vec<float, 1>>);
+static_assert(              !std::is_assignable_v<vec<half, 2>, vec<double, 1>>);
+static_assert(               std::is_assignable_v<vec<half, 2>, sw_half_1>);
+static_assert(               std::is_assignable_v<vec<half, 2>, sw_float_1>);
+static_assert(               std::is_assignable_v<vec<half, 2>, sw_double_1>);
+static_assert(               std::is_assignable_v<vec<half, 2>, sw_half_2>);
+static_assert(              !std::is_assignable_v<vec<half, 2>, sw_float_2>);
+static_assert(              !std::is_assignable_v<vec<half, 2>, sw_double_2>);
+
+static_assert(               std::is_assignable_v<vec<float, 1>, half>);
+static_assert(               std::is_assignable_v<vec<float, 1>, float>);
+static_assert(               std::is_assignable_v<vec<float, 1>, double>);
+#if __SYCL_DEVICE_ONLY__
+static_assert(               std::is_assignable_v<vec<float, 1>, vec<half, 1>>);
+#else
+static_assert(NOT_IN_PREVIEW std::is_assignable_v<vec<float, 1>, vec<half, 1>>);
+#endif
```

Comment:
Sorry, could you remind me why this behavior should differ on host vs. device?

Reply:
From the PR's description:

> Note that the "test" added here is less of a documentation of "intent" but rather the documentation of the effect of the change.

The main reason is `vector_t`, I think. It's going to be deleted in another PR (haven't uploaded it yet).

Reply:
Also, I'm adding `PREVIEW_ONLY` in https://github.com/intel/llvm/pull/17713/commits/93a3f7f568ec64b162fee4831d02619ea70c379b, which, IMO, complements `NOT_IN_PREVIEW` nicely.

Reply:
Cool